components/provisioning/warewulf-provision/SOURCES/warewulf-provision.pxe_file_modes.patch

```diff
@@ -36,35 +45,64 @@
              print COOKIE $self->checksum();
              close COOKIE;
              &nprint("Bootstrap image '$bootstrap_name' is ready\n");
---- a/lib/Warewulf/Provision/Pxe.pm	2019-06-17 14:01:40.000000000 -0700
-+++ b/lib/Warewulf/Provision/Pxe.pm	2019-06-25 14:42:31.414866655 -0700
-@@ -94,8 +94,11 @@
+diff --git a/lib/Warewulf/Provision/Pxe.pm b/lib/Warewulf/Provision/Pxe.pm
+index c9f6a7b..c843206 100644
+--- a/lib/Warewulf/Provision/Pxe.pm
++++ b/lib/Warewulf/Provision/Pxe.pm
+@@ -19,6 +19,7 @@ use Warewulf::DataStore;
+ use Warewulf::Provision::Tftp;
+ use File::Basename;
+ use File::Path;
++use File::Path qw(make_path);
+ use POSIX qw(uname);
+ 
+ our @ISA = ('Warewulf::Object');
+@@ -94,8 +95,11 @@ setup()
                  if (-f "$datadir/warewulf/ipxe/$f") {
                      &iprint("Copying $f to the tftp root\n");
                      my $dirname = dirname("$tftpdir/warewulf/ipxe/$f");
 -                    mkpath($dirname);
-+                    mkpath($dirname, {
-+                        mode => 0755,
++                    make_path($dirname, {
++                        chmod => 0755,
 +                    });
                      system("cp $datadir/warewulf/ipxe/$f $tftpdir/warewulf/ipxe/$f");
 +                    chmod 0644, "$tftpdir/warewulf/ipxe/$f";
                  } elsif ($arch eq "x86_64") {
                      &eprint("Could not locate Warewulf's internal $datadir/warewulf/ipxe/$f! Things might be broken!\n");
                  }
-@@ -106,8 +109,11 @@
+@@ -106,8 +110,11 @@ setup()
                  if (-f "$datadir/warewulf/ipxe/$f") {
                      &iprint("Copying $f to the tftp root\n");
                      my $dirname = dirname("$tftpdir/warewulf/ipxe/$f");
 -                    mkpath($dirname);
-+                    mkpath($dirname, {
-+                        mode => 0755,
++                    make_path($dirname, {
++                        chmod => 0755,
 +                    });
                      system("cp $datadir/warewulf/ipxe/$f $tftpdir/warewulf/ipxe/$f");
 +                    chmod 0644, "$tftpdir/warewulf/ipxe/$f";
                  } elsif ($arch eq "aarch64") {
                      &eprint("Could not locate Warewulf's internal $datadir/warewulf/ipxe/$f! Things might be broken!\n");
                  }
-@@ -372,6 +380,9 @@
+@@ -154,7 +161,9 @@ update()
+ 
+     if (! -d "$statedir/warewulf/ipxe/cfg") {
+         &iprint("Creating ipxe configuration directory: $statedir/warewulf/ipxe/cfg");
+-        mkpath("$statedir/warewulf/ipxe/cfg");
++        make_path("$statedir/warewulf/ipxe/cfg", {
++            chmod => 0755,
++        });
+     }
+ 
+     foreach my $nodeobj (@nodeobjs) {
+@@ -322,6 +331,7 @@ update()
+                     if (! close IPXE) {
+                         &eprint("Could not write iPXE configuration file: $!\n");
+                     }
++		    chmod 0644, "$statedir/warewulf/ipxe/cfg/$config";
```

Comment:
```suggestion
+		    chmod(0644, "$statedir/warewulf/ipxe/cfg/$config");
```

This fits closer too where we're using `chmod` in a few other places.

Reply:
Added this to all the chmod incantations.